src/modules/iotjs_module_process.c

```diff
@@ -78,24 +78,33 @@ static jerry_value_t WrapEval(const char* name, size_t name_len,
 
 
 JS_FUNCTION(Compile) {
-  DJS_CHECK_ARGS(2, string, string);
-
-  iotjs_string_t file = JS_GET_ARG(0, string);
-  iotjs_string_t source = JS_GET_ARG(1, string);
+  DJS_CHECK_ARGS(1, string);
 
-  const char* filename = iotjs_string_data(&file);
+  iotjs_string_t path = JS_GET_ARG(0, string);
   const iotjs_environment_t* env = iotjs_environment_get();
-
   if (iotjs_environment_config(env)->debugger != NULL) {
     jerry_debugger_stop();
   }
 
-  jerry_value_t jres =
-      WrapEval(filename, strlen(filename), iotjs_string_data(&source),
-               iotjs_string_size(&source));
+  const char* filename = iotjs_string_data(&path);
+  uv_fs_t fs_req;
+  uv_fs_stat(iotjs_environment_loop(env), &fs_req, filename, NULL);
+  uv_fs_req_cleanup(&fs_req);
+
+  if (!S_ISREG(fs_req.statbuf.st_mode)) {
+    iotjs_string_destroy(&path);
+    return JS_CREATE_ERROR(COMMON, "ReadSource error, not a regular file");
+  }
+
+  size_t size = 0;
+  char* source = iotjs__file_read(iotjs_string_data(&path), &size);
```

Comment:
Here and following previously extracted `filename` could be used for convenience and destroy `path` right after LINE 89. :p